src/lib/functions/registry.js

```diff
@@ -102,18 +102,7 @@ class FunctionsRegistry {
     return this.functions.get(name)
   }
 
-  registerFunction(name, func) {
-    if (func.isBackground && !this.capabilities.backgroundFunctions) {
-      this.logger.warn(getLogMessage('functions.backgroundNotSupported'))
-    }
-
-    this.functions.set(name, func)
-    this.buildFunctionAndWatchFiles(func)
-
-    this.logger.log(`${NETLIFYDEVLOG} ${chalk.green('Loaded')} function ${chalk.yellow(name)}.`)
-  }
-
-  async scan(directory) {
+  async prepareDirectoryScan(directory) {
```

Comment:
`prepareDirectoryScan` is now responsible for creating the directory and running any `onDirectoryScan` hooks. It runs at the beginning of `scan` for every configured directory.